handsontable/src/plugins/customBorders/utils.js

```diff
@@ -48,13 +50,53 @@ export function createDefaultHtBorder() {
   };
 }
 
+/**
+ * Normalizes the border object to be compatible with the Border API from the Walkontable.
+ * The function translates the "left"/"right" properties to "start"/"end" prop names.
+ *
+ * @param {object} border The configuration object of the border.
+ * @returns {object}
+ */
+export function normalizeBorder(border) {
+  return {
+    row: border.row,
+    col: border.col,
+    ...(isDefined(border.id) ? { id: border.id } : null),
+    ...(isDefined(border.border) ? { border: border.border } : null),
+    ...(isDefined(border.top) ? { top: border.top } : null),
+    ...(isDefined(border.bottom) ? { bottom: border.bottom } : null),
+    ...(isDefined(border.start) || isDefined(border.left) ? { start: border.start ?? border.left } : null),
+    ...(isDefined(border.end) || isDefined(border.right) ? { end: border.end ?? border.right } : null),
+  };
+}
+
+/**
+ * Denormalizes the border object to be backward compatible with the previous version of the CustomBorders
+ * plugin API. The function extends the border configuration object for the backward compatible "left"/"right"
+ * properties.
+ *
+ * @param {object} border The configuration object of the border.
+ * @returns {object}
+ */
+export function denormalizeBorder(border) {
+  return {
+    row: border.row,
+    col: border.col,
+    ...(isDefined(border.id) ? { id: border.id } : {}),
+    ...(isDefined(border.border) ? { border: border.border } : {}),
+    ...(isDefined(border.top) ? { top: border.top } : {}),
+    ...(isDefined(border.bottom) ? { bottom: border.bottom } : {}),
+    ...(isDefined(border.start) ? { start: border.start, left: border.start } : {}),
+    ...(isDefined(border.end) ? { end: border.end, right: border.end } : {}),
+  };
+}
+
```

Comment:
Both operations could have a performance impact because of many `Object.assigns`. Much more convenient seems to be using stable numbers of properties.

These also create new objects which have different references - it is a design decision about is it ok with the current architecture when probably some changes can be provided through references? I'm not sure what the current workflow about borders looks like, so let's be careful here while testing.

Reply:
Thanks for pointing that out. Generally, I'm aware of that. The decision here is dictated by the fact that the plugin is not very well written. It passes the objects by reference between different methods, modifying some data (by ref), checking some props ... all of that causes a lot of confusion. As if that was not enough, some of the objects are modified by reference from the Walkontable Border class - so basically a very bad idea.

The refactor of the plugin would be out of the scope of the task.

Reply:
In general, your implementation creates new objects so it loses the references to the main Border object. That I tried to point out. In your message, you confirm that keeping references is required, so creating a new object each time when calling normalize and denormalize is incorrect.